### PR TITLE
Design your own framework

### PR DESCRIPTION
**Pull request by Israel-Laguan**
# Resume

We reviewed the site and compare it to the original, also check the stickler linter, W3C validators, microverse requirements, and *came to the conclusion that we met all of the requirements*.

# Proposal

We want to have a review of the code to know if we need to make any changes. If no changes needed we want the project to be ACCEPTED so we can move to the next project with a happy smile :)

# Comments

Making the grid system was fun, regardless of booth of us thinking that making that same with `flex-basis` or `grid` is faster. But in general is fine to have that tool when IE support strike  us later.

### CODE REVIEW

**Comment by ebukaume** [comment]
Hi @Israel-Laguan  and @MarioDena :wave: 

Great job. You guys have really done well. :+1:  I like your commit messages and PR description. :fireworks: Guys, your README is one of the best I have seen. Please keep the good work up.

However, let's fix these issues below:

**Could you have done this differently? Have a look at [this article](http://thesassway.com/intermediate/if-for-each-while#for)**
https://github.com/Israel-Laguan/Design-your-own-Framework/blob/c26489adef02ac12bb5e9af6d72bb93af63f7ee0/css/_grid.scss#L21-L38

**Any reason you guys used fixed widths here? Why not use %? That way, the elements scale smoothly. It would also be more understandable and thus easily maintainable**
https://github.com/Israel-Laguan/Design-your-own-Framework/blob/c26489adef02ac12bb5e9af6d72bb93af63f7ee0/css/_grid.scss#L39-L85

**Which website did you guys clone?**
You may want to read [this part of the requirements well](https://www.theodinproject.com/courses/html5-and-css3/lessons/design-your-own-grid-based-framework#applying-the-framework). You can ignore No 1. on the list and use this same repo.
Also, I can't seem to find a class that represents a row.


[OPTIONAL]
You can add some utility classes to your framework. Things like margin, padding, text, color and positioning utility classes. You can refer to [this](https://getbootstrap.com/docs/4.0/utilities/text/) for inspirations.


**Kindly submit another code review request after fixing the above issues**


Happy Coding :v: 
**Ebuka Umeokonkwo**
 [Github](https://github.com/ebukaume) | [Linkedin](https://www.linkedin.com/in/ebukaume) | [Slack](https://app.slack.com/team/UFDKXE68L) | [Twitter](https://twitter.com/ebukaume)

**Comment by Israel-Laguan** [comment]
Thanks for your review @ebukaume !! OK, lets address your points: 

`const [ reviewPoints, optionalPoints] = [` 
> **Could you have done this differently?** `,`

> **Any reason you guys used fixed widths here? Why not use %?** `,`

> **Which website did you guys clone?**`,`

> Also, I can't seem to find a class that represents a row.

`], [`

> You can add some utility classes to your framework. 

`];`

I think that the `reviewPoints[0]` and `reviewPoints[1]`, are kinda fair but is more like optional, we can check that anyway.  We can search a site to clone so we can address `reviewPoints[2]`.

 `reviewPoints[3]` is just a matter of names or not really necessary because 2 or more `divs` that use _12 `column`s_ or less is a `row`, like `row` not really needs to be a _class_ but a _logical abstraction_. For example, what about making a lot of `row`s for a gallery when is easier and even more understandable just saying the number of column from the grid for each image to take? or if I have 3 images that takes 6 columns each on a grid of 12 columns but those 3 images are under a `row` `div`?

About `optionalPoints[0]`, utilities, is a great suggestion.

Thanks for your feedback!

**Comment by MarioDena** [comment]
Added a desired clone example to the grid.

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Hi team!
You have done a good Job. However, in this project, the result is to come up with css framework like bootsrap. Having gone through your css and sass file, it is missing some crucial implementation.

- [x] A class that represents a row

 - [x] Classes to define columns sized from 1/12th of the row to 100% of the row

 - [x]  The defined columns should have breakpoints (example column-large-2)

Please, use this [project](https://github.com/b-a-a-s/mv-design-your-own-grid-based-framework/tree/design-your-own-grid-based-framework) as a guide. :slightly_smiling_face: 

After making the needed updates, submit another form for a code review. 

#### ``` Have a great day ``` :slightly_smiling_face: 

@ Uchechukwu Henry Inyama

**Comment by MarioDena** [comment]
@uche-inyama 

None of that is part of the project requierements either here or on the microverse pathwrigth.

A quote from the projects specifications:  The complexity is up to you – there’s no need to go too crazy with details.

Please read here: https://www.theodinproject.com/courses/html5-and-css3/lessons/design-your-own-grid-based-framework

**Comment by MarioDena** [comment]
@uche-inyama Hello! thank you very much for your time! Sadly, none of what you asked for is part of the project specifications. 

1. The whole grid file has breakpoints.
2. We don't need a row class. Container_12 is a row... Or if you want a pure row you have our wrapper class.
3. Container1-12 are classes that define columns...

Your project guide is not official we are using THIS: https://www.theodinproject.com/courses/html5-and-css3/lessons/design-your-own-grid-based-framework and THIS: https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59564673/

On a side note cheers! to @ebukaume  Thank you for referring us to the SASS for feature, I'm still reading the page.

File: index.html [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      </div>
+      <div class="container_8 spacer">
+        space
+      </div>
+      <div class="container_8 spacer">
+        space
+      </div>
+      <div class="container_4 aside">
+        <div class="floatRight">
+          <div class="customButton"><p><</p></div>
```
Comment (Redvanisation) [inline index.html]: You have a markup validation error in this line, you should delete the extra `<` in the `<p>` tag.

**Review by Redvanisation — APPROVED** [review]
### Hello team 🌟 

### _Nice work on this project, it seems that you guys have met the requirements so I am approving your project. 👍_ 

**However, fix the validation error mentioned in my comment below first, then ⬇️** 

### You may merge to master and submit a project completion form. 🚀 

### Good luck! 🍀

File: index.html (hunk 1 above)
Comment (MarioDena) [reply index.html]: It was a nasty error on my part should have used the code for "<" instead of the character, fixing it in the morning.

**Comment by MarioDena** [comment]
@Redvanisation I should have used the markup code instead of the character for the button, nice catch, project updated, HTML re-validated and branch merged, thank you for your input.

**Pull request merged**